components/providers/provider-detail-client.tsx

```
@@ -141,6 +142,37 @@ export function ProviderDetailClient({ providerId }: ProviderDetailClientProps)
+    const { hasReviewedProvider, isLoading } = useHasReview(providerId);
+    
+    if (isLoading) {
+      return <div className="flex justify-center py-4"><Loader2 className="h-6 w-6 animate-spin text-primary" /></div>;
+    }
+    
+    if (hasReviewedProvider) {
+      return (
+        <div className="flex items-center gap-3 p-4 bg-muted/50 rounded-md">
+          <CheckCircle className="h-5 w-5 text-primary" />
+          <div>
+            <p className="font-medium">You've already reviewed this provider</p>
```

ESLint

[error] `'` can be escaped with `&apos;`, `&lsquo;`, `&#39;`, `&rsquo;`.

(react/no-unescaped-entities)